ipatests/test_integration/test_installation.py

```diff
@@ -570,22 +570,90 @@ class TestInstallWithCA_DNS3(CALessBase):
     ticket 7239
     """
 
+    @pytest.mark.skipif(
+        osinfo.id == 'fedora',
+        reason='https://pagure.io/freeipa/issue/8700',
+        strict=True
+    )
     @server_install_setup
     def test_number_of_zones(self):
         """There should be two zones: one forward, one reverse"""
 
         self.create_pkcs12('ca1/server')
         self.prepare_cacert('ca1')
 
-        self.install_server(extra_args=['--allow-zone-overlap'])
+        # Clean NM DNS configuration.
+        # Then apply revolved.
+        if osinfo.id == 'fedora' and osinfo.version_number >= (34,):
+
+            nmcli_script_file = "/root/nmcli_script"
+            nmcli_script = ("nmcli -g name,type connection  show  --active "
+                            "| awk -F: '/ethernet|wireless/ { print $1 }'"
+                            "| while read connection\n"
+                            "do\n"
+                            "  nmcli con mod \"$connection\" "
+                            "ipv6.ignore-auto-dns yes\n"
+                            "  nmcli con mod \"$connection\" "
+                            "ipv4.ignore-auto-dns yes\n"
+                            "  nmcli con down \"$connection\" "
+                            "&& nmcli con up \"$connection\"\n"
+                            "done\n")
+            self.master.put_file_contents(nmcli_script_file , nmcli_script)
+            self.master.run_command(["bash", nmcli_script_file])
+
+            # remove all resolved netif-specific configuration
+            result = self.master.run_command(
+                ["ls", "/run/systemd/resolve/netif/"]
+            )
+            for netif in result.stdout_text.splitlines():
+                self.master.run_command(
+                    ["resolvconf", "-d", f"{netif}"]
+                )
 
-        result = self.master.run_command([
-            'ipa', 'dnszone-find'])
+            # force using 127.0.0.2
+            self.master.run_command(
```

Comment:
In current version this is already done by lines
```
master.resolver.setup_resolver('127.0.0.2')
```

It does it in a more preferred way by dropping config file in `/etc/systemd/resolved.conf.d/`